include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -4722,6 +4722,24 @@ struct OrtApi {
    * \param[in] adapter OrtLoraAdapter instance
    */
   ORT_API2_STATUS(RunOptionsAddActiveLoraAdapter, _Inout_ OrtRunOptions* options, _In_ const OrtLoraAdapter* adapter);
+
+  /// @}
+  /// \name OrtEpDynamicOptions
+  /// @{
+
+  /** \brief Set DynamicOptions for EPs
+   *
+   * \param[in] session
+   * \param[in] list of keys represented by null-terminated strings
+   * \param[in] list of values represented by null-terminated strings
+   * \param[in] number of key-value pairs
+   *
+   * \since Version xxx
+   * @TODO: update version number
+   * @TODO: should it be SetExecutionProvider... instead of SetEp...?
+   */
+  ORT_API2_STATUS(SetEpDynamicOptions, _In_ OrtSession* sess, _In_ const char** keys, _In_ const char** values,
```

Comment:
ORT_API2_STATUS(SetEpDynamicOptions, Inout OrtSession* sess, In_reads(kv_len) const char* const* keys, In_reads(kv_len) const char* const* values, In size_t kv_len)

Reply:
formatting looks off